nautobot/core/views/generic.py

```diff
@@ -379,7 +376,7 @@ def get_required_permission(self):
     def get_object(self, kwargs):
         """Retrieve an object based on `kwargs`."""
         # Look up an existing object by slug, PK, or name, if provided.
-        for field in ("slug", "pk", "name"):
+        for field in ("slug", "pk", "name", "key"):
             if field in kwargs:
                 return get_object_or_404(self.queryset, **{field: kwargs[field]})
```

Comment:
Longer term will be great to replace this with something based on `natural_key_slug`...